subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -280,6 +280,11 @@ static int download_client_callback(const struct download_client_evt *event)
 			/* Fall through and return 0 below to tell
 			 * download_client to retry
 			 */
+		} else if ((event->error == -ECONNABORTED) || (event->error == -ECONNREFUSED)) {
+			LOG_ERR("Download client failed to connect to server");
+			set_error_state(FOTA_DOWNLOAD_ERROR_CAUSE_CONNECT_FAILED);
+
+			goto error_and_close;
```

Comment:
Can this be combined with the `else` case below?
Or why do we not need to call `dfu_target_done()` here, but we do below.

Reply:
`dfu_target_done(false)` is called also under the `error_and_close` label, so should not need to call it here too. I can instead remove it below in the `else`

Reply:
Oh, I see.
Then only difference is the verbosity, which I don't think is relevant.. I'm fine with how it is.